src/python/pants/source/wrapped_globs.py

```diff
@@ -4,37 +4,45 @@
 import os
 from abc import ABC, abstractmethod
 from hashlib import sha1
+from typing import Callable, Iterable, Iterator, List, Sequence, Tuple, Union, cast
 
 from twitter.common.dirutil.fileset import Fileset
+from typing_extensions import TypedDict
 
 from pants.base.build_environment import get_buildroot
-from pants.engine.fs import EMPTY_SNAPSHOT
+from pants.base.parse_context import ParseContext
+from pants.engine.fs import EMPTY_SNAPSHOT, Snapshot
 from pants.util.dirutil import fast_relpath, fast_relpath_optional
 from pants.util.memo import memoized_property
 
 
+class _GlobsDict(TypedDict, total=True):
+  globs: List[str]
```

Comment:
This means that we expect this dict: `{"globs": ['foo', 'bar']`. MyPy will complain with anything else like `{}` or `{"bad": 1}`.